sdc/hiframes/pd_dataframe_ext.py

```diff
@@ -66,6 +72,29 @@ def init_dataframe(typingctx, *args):
     index_typ = args[n_cols]
     column_names = tuple(a.literal_value for a in args[n_cols + 1:])
 
+    # Define df structure, map column name to column position ex. {'A': (0,0), 'B': (1,0), 'C': (0,1)}
+    df_structure = {}
+    # Store unique types of columns ex. {'int64': (0, [0, 2]), 'float64': (1, [1])}
+    data_typs_map = {}
+    types_order = []
+    type_id = 0
+    for col_id, col_typ in enumerate(data_typs):
+        col_name = column_names[col_id]
+
+        if col_typ not in data_typs_map:
+            data_typs_map[col_typ] = (type_id, [col_id])
+            # The first column in each type always has 0 index
+            df_structure[col_name] = ColumnId(type_id, 0)
+            types_order.append(col_typ)
+        else:
+            # Get index of column in list of types
+            type_id, col_indices = data_typs_map[col_typ]
```

Comment:
`type_id` looks like a counter. Is it ok that we rewrite it here? Maybe use another name for `type_id` in else-block and increment `type_id` only inside of if-block.